### ISSUE TITLE
Cop incorrectly handles missing space after = in endless functions

### ISSUE THREAD

**Issue by Stiwyy** [rubocop/rubocop#15562]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False negative

### Cop name

Layout/SpaceAroundOperators

### Steps to reproduce the problem

1. create a an example ruby file.
2. define an endless function with no space after the = operator like this: 
```ruby 
def hello ='hello world'
```
3. run rubocop example.rb

### Expected behavior

After running 
```shell
rubocop example.rb
```
Rubocop should detect an offense: 
`Layout/SpaceAroundOperators: Surrounding space missing for operator =.`

Running 
```shell
rubocop example.rb -a
```
should change example.rb to:
```diff
-def hello ='hello world'
+def hello = 'hello world'
```

### Actual behavior

After running rubocop example.rb, Rubocop doesn't detect an offense.

### RuboCop configuration

```yaml
AllCops:
  NewCops: enable
  TargetRubyVersion: 4.0
```

### RuboCop version

```shell
1.89.0 (using Parser 3.3.12.0, Prism 1.9.0, rubocop-ast 1.50.0, analyzing as Ruby 4.0, running on ruby 4.0.6) +server [arm64-darwin23]
```
